scripts/setup-minikube.sh

```diff
@@ -0,0 +1,354 @@
+  read v
+  if [[ $v != "" ]]
+  then
+    MINIKUBE_CMD=$MINIKUBE_CMD" --vm-driver=$v"
+  fi
+
+  # Mount local dir
+  echo -n "If you'd like to access a local directory in JupyterHub, please enter the full path: "
+  read l
+  if [[ $l != "" ]]
+  then
+    MINIKUBE_CMD=$MINIKUBE_CMD" --mount-string=$l:/mnt/local --mount"
+    MOUNT_LOCAL=true
+  fi
+}
+
+function download_kfctl_scripts() {
+  curl -O https://raw.githubusercontent.com/kubeflow/kubeflow/master/scripts/kfctl.sh
+  chmod +x kfctl.sh
+  curl -O https://raw.githubusercontent.com/kubeflow/kubeflow/master/scripts/util.sh
+  mkdir -p gke
+  pushd .
+  cd gke
+  curl -O https://raw.githubusercontent.com/kubeflow/kubeflow/master/scripts/gke/util.sh
+  popd
+}
+
+function deploy_kubeflow() {
+  # pull bootstrapper config and deloy kubeflow
+  # curl -O https://raw.githubusercontent.com/kubeflow/kubeflow/v0.2.0/bootstrap/bootstrapper.yaml
+  # kubectl create -f bootstrapper.yaml
+
+  # kfctl installation
+  set +e
+  O=`kubectl get namespace kubeflow 2>&1`
+  RESULT=$?
+  set -e
+
+  if [ "${RESULT}" -eq 0 ]; then
+    echo namespace kubeflow already exists
+  else
+    kubectl create namespace kubeflow
+  fi
+
+  #download_kfctl_scripts
+  KUBEFLOW_REPO=/usr/local/google/home/abhishek/kubeflow ./kfctl.sh init localapp --platform minikube
+  pushd .
+  cd localapp
+  ../kfctl.sh generate all
+  ../kfctl.sh apply all
+  popd
+
+  if is_kubeflow_ready
+  then
+    mount_local_fs
+    setup_tunnels
+  else
+    echo -e "${RED}Unable to get kubeflow ready${NC}"
+  fi
+}
+
+function is_kubeflow_ready() {
+#  echo -ne "${YELLOW}Downloading bootstrapper image."
+#  until [[ `kubectl -n kubeflow-admin get pod kubeflow-bootstrapper-0 | tail -1 | awk '{print $3}'` == "Running" ]]
+#  do
+#    echo -n "."
+#    sleep 30
+#  done
+#  echo -e "${GREEN}[OK]${NC}"
+
+  echo -en "${YELLOW}Getting kubeflow namespace ready...${NC}"
+  local ns_ready=false
+  for i in {1..5}
+  do
+    kube_ns=`kubectl get namespaces | grep kubeflow | wc -l`
+    # if [ $kube_ns = 2 ]
+    if [ $kube_ns = 1 ]  # for kfctl
+    then
+       ns_ready=true
+       echo -e "${GREEN}[OK]${NC}"
+       break
+    fi
+    sleep 30
+  done
+
+  if [[ $ns_ready = false ]]
+  then
+    return 1
+  fi
+
+  svc_ready=false
+  amb_up=0
+  tf_hub_up=0
+  echo -en "${YELLOW}Bringing kubeflow services up."
+  until (( "$amb_up" > 0 && "$tf_hub_up" > 0 ))
+  do
+    sleep 30
+    amb_up=`kubectl -n kubeflow get pods | grep Running | grep ambassador | wc -l`
+    tf_hub_up=`kubectl -n kubeflow get pods | grep Running | grep tf-hub | wc -l`
+    echo -n "."
+    if (( "$amb_up" > 0 && "$tf_hub_up" > 0 ))
+    then
+      svc_ready=true
+      echo -e "${GREEN}[OK]${NC}"
+      break
+    fi
+  done
+
+  if [ $svc_ready = false ]
+  then
+    return 1
+  fi
+}
+
+
+# if user requested a local fs path to be mounted, make it accessible via
+# Jupyter Notebooks
+function mount_local_fs() {
```

Comment:
To better fit this into the kfctl.sh pattern;
pv.yaml should be dumped to the k8s_specs directory during the generate phase of the kfctl command. And it should then be created during the kfctl apply phase